Vagrantfile

```diff
@@ -15,9 +15,27 @@ Vagrant.configure("2") do |config|
   config.vm.provision "shell", inline: <<-SHELL
     cat << EOF | dnf -y shell
 update
-install podman make golang-go libseccomp-devel bats jq
+install podman make golang-go libseccomp-devel bats jq \
+ patch protobuf protobuf-c protobuf-c-devel protobuf-compiler \
+ protobuf-devel protobuf-python libnl3-devel libcap-devel libnet-devel \
+ nftables-devel libbsd-devel gnutls-devel
 ts run
 EOF
     dnf clean all
+
+    # TODO: remove this after criu 3.14 is released
+    rpm -e --nodeps criu || true
+    CRIU_VERSION=v3.13
+    mkdir -p /usr/src/criu \
+        && curl -fsSL https://github.com/checkpoint-restore/criu/archive/${CRIU_VERSION}.tar.gz | tar -C /usr/src/criu/ -xz --strip-components=1 \
+        && cd /usr/src/criu \
+        && echo 1 > .gitid \
+        && curl -sSL https://github.com/checkpoint-restore/criu/commit/4c27b3db4f4325a311d8bfa9a50ea3efb4d6e377.patch | patch -p1 \
+        && curl -sSL https://github.com/checkpoint-restore/criu/commit/aac41164b2cd7f0d2047f207b32844524682e43f.patch | patch -p1 \
+        && curl -sSL https://github.com/checkpoint-restore/criu/commit/6f19249b2565f3f7c0a1f8f65b4ae180e8f7f34b.patch | patch -p1 \
+        && curl -sSL https://github.com/checkpoint-restore/criu/commit/378337a496ca759848180bc5411e4446298c5e4e.patch | patch -p1 \
+        && make install-criu \
+        && cd - \
+        && rm -rf /usr/src/criu
```

Comment:
Why not in Dockerfile?

Reply:
Yes, we have the very same code in Dockerfile (this is mostly a copy-paste).

We are running these tests on the Vagrant host itself (not inside a container, i.e. `make localintegration` not `make integration`) and some of those tests use criu, so we have to have criu on the vagrant host, not inside a container.

Or do you mean we can copy the criu binary out of container we just built and use it? This is also possible (and I tried to do it with runc binary earlier, and you suggested to rebuild it on the host itself, so I assumed you have the same PoV about criu binary).

As noted in the comment, this will go away as soon as criu 3.14 is released and packaged for F31.

Reply:
If we want to run everything on Vagrant directly, can we eliminate Podman from Vagrantfile? Can be another PR though.

Reply:
Yes, we can followup on that. This just fixes the cgroupv2 checkpoint and covers it with tests.

See https://github.com/opencontainers/runc/pull/2342. Note the difference is we used to run unit tests on Debian (container) and now it's Fedora (host).